File: scripts/R/plot_exon_counts.R
data <- read.table("results/exon_counts_final.tsv", header=TRUE, sep="\t")

data$Short_ID <- sub("TCONS_000", "", data$Transcript)
cols <- ifelse(data$Gene == "Ano1", "steelblue", "orange")

png("results/figures/exon_counts_prioritised_Ano1_Ano8.png",
    width=1800, height=1100, res=300)

bp <- barplot(data$Exons,
              names.arg=data$Short_ID,
              col=cols,
              ylim=c(0, 32),
              main="Exon counts of prioritised Ano1 and Ano8 transcript isoforms",
              xlab="Candidate transcript",
              ylab="Number of exons")

text(bp, data$Exons + 1, labels=data$Exons, cex=0.8)

legend("topright",
       legend=c("Ano1", "Ano8"),
       fill=c("steelblue", "orange"),
       bty="n")

dev.off()

File: scripts/python/plot_exon_counts.py
import matplotlib.pyplot as plt
from matplotlib.patches import Patch

genes = ["Ano1","Ano1","Ano1","Ano1","Ano1","Ano8","Ano8","Ano8"]
transcripts = [
    "00004856","00004857","00004858","00008054","00008053",
    "00026007","00025093","00026006"
]
exons = [26,27,28,27,26,17,16,15]

colors = ["steelblue" if g == "Ano1" else "orange" for g in genes]

plt.figure(figsize=(8,4.5))
bars = plt.bar(transcripts, exons, color=colors)

plt.title("Exon counts of prioritised Ano1 and Ano8 transcript isoforms")
plt.xlabel("Candidate transcript")
plt.ylabel("Number of exons")

for bar, value in zip(bars, exons):
    plt.text(
        bar.get_x() + bar.get_width()/2,
        value + 0.3,
        str(value),
        ha="center",
        va="bottom",
        fontsize=9
    )

legend_elements = [
    Patch(facecolor="steelblue", label="Ano1"),
    Patch(facecolor="orange", label="Ano8")
]
plt.legend(handles=legend_elements, frameon=False)

plt.tight_layout()
plt.savefig("results/figures/exon_counts_prioritised_Ano1_Ano8.png", dpi=300)
plt.savefig("results/figures/exon_counts_prioritised_Ano1_Ano8.pdf")
print("Saved:")
print("results/figures/exon_counts_prioritised_Ano1_Ano8.png")
print("results/figures/exon_counts_prioritised_Ano1_Ano8.pdf")
